src/models/blockchain/block/mod.rs

```diff
@@ -2047,44 +2050,227 @@ pub(crate) mod tests {
-            //     &b_previous, now, Network::Main
-            // ).await.err().unwrap()];
-
-            /* proof requiring error will be reached with `TestRunner::deterministic()`
-            // #[traced_test]
-            #[apply(shared_tokio_runtime)]
-            async fn test_negs() {
-                todo!("BlockValidationError::TransactionTimestamp");
-
-                let mut test_runner = TestRunner::deterministic();
-                let (b_prev, ts_next) = strategies::previous().prop_flat_map(|b_prev| {
-                    let ts_prev = b_prev.kernel.header.timestamp.0.value();
-                    (proptest::prelude::Just(b_prev), (ts_prev + 6000)..(ts_prev + 1288490188500000))
-                }).new_tree(&mut test_runner).unwrap().current();
-                // struct Test {
-                //     b_prev: Block,
-                //     b_next: Block,
-                //     now: u64,
-                //     network: Network,
-                // };
-                let ts_next = Timestamp(bfe![ts_next]);
-
-                let mut rng_seeded = StdRng::seed_from_u64(2225550001);
-                let b_next =
-                    loop {
-                        let b_n = fake_valid_successor_for_tests(
-                            &b_prev, ts_next, rng_seeded.random(), Network::Main
-                        ).await;
-                        if !dbg!(b_n.kernel.body.transaction_kernel.fee).is_negative() {break b_n;}
-                    };
-                // prop_assume![!dbg!(b_next.kernel.body.transaction_kernel.fee).is_negative()];
-
-                // see `shared::fake_create_block_transaction_for_tests` for using `nop`
-
-                // dbg![b_next.validate(&b_now, ts_next, Network::Main).await];
-            } */
+            // Test AppendixTooLarge
+            // let mut b_now_large_appendix = b_now.clone();
+            // let large_claims = vec![Claim::default(); MAX_NUM_CLAIMS + 1];
+            // b_now_large_appendix.kernel.appendix = BlockAppendix::new(large_claims);
+            // prop_assert_eq![
+            //     BlockValidationError::AppendixTooLarge,
+            //     b_now_large_appendix
+            //         .validate(&b_previous, now, Network::Main)
+            //         .await
+            //         .err()
+            //         .unwrap()
+            // ];
+
+            let b_now_timestamp_header = b_now.kernel.header.timestamp;
+            let mut tx_kernel =
+                TransactionKernelProxy::from(b_now.kernel.body.transaction_kernel.clone());
+            if b_now.kernel.body.transaction_kernel.timestamp > b_now_timestamp_header {
+                tx_kernel.timestamp = b_now_timestamp_header;
+            }
+            tx_kernel.inputs = Vec::new();
+            b_now.kernel.body.transaction_kernel = tx_kernel.into_kernel();
+            b_now.kernel.appendix =
+                BlockAppendix::new(BlockAppendix::consensus_claims(b_now.body()));
```

Comment:
Work in progress, I guess? I'm not sure if this block of code belongs to the previous (AppendixTooLarge) or next (CoinbaseTooBig).